truffle/com.oracle.truffle.api.vm/src/com/oracle/truffle/api/vm/PolyglotEngine.java

```diff
@@ -265,6 +270,29 @@ public Builder onEvent(EventConsumer<?> handler) {
         }
 
         /**
+         * Provide configuration data to initialize the {@link PolyglotEngine} for a specific
+         * language. These arguments can be used by languages to initialize and configure their
+         * initial execution state.
```

Comment:
It may be good idea to connect the method to the getConfig one. See
https://github.com/jtulach/truffle/blob/32d4240880fd2a97100c3eb1776e274d7c8bbfbb/truffle/com.oracle.truffle.api.vm/src/com/oracle/truffle/api/vm/PolyglotEngine.java#L263